src/endpoint/blob/blob_utils.js

```diff
@@ -64,12 +64,13 @@ function parse_copy_source(req) {
 async function list_objects(params, account_name, container, sasToken) {
 
     const hostname = `${account_name}.blob.core.windows.net`;
-    const delimiter = params.delimiter || '/';
     const maxResults = params.limit === 0 ? 1 : params.limit;
```

Comment:
This is incorrect.
First of all the default is 5000 and not 1.
Secondly, setting maxresults to a value less than or equal to zero results in error response code 400 (Bad Request).
It is a bug to check for zero and assign a 1.

Reply:
it's not a part of my changes but ok, 
If i'm not mistaken it's not a check for default and the reason behind it it's for handling maxResults=0 is allowed on AWS and in Blob ```Setting maxresults to a value less than or equal to zero results in error response code 400 (Bad Request).```

Reply:
In any case, this isn't compatible.
`(base) jeniawhite@MacBook-Pro noobaa-core-code % aws s3api list-objects --bucket jenia --max-keys 0
{
    "IsTruncated": false,
    "Marker": "",
    "Name": "jenia",
    "Prefix": "",
    "MaxKeys": 0,
    "EncodingType": "url"
}
(base) jeniawhite@MacBook-Pro noobaa-core-code % aws s3api list-objects --bucket jenia --max-keys 1
{
    "IsTruncated": true,
    "Marker": "",
    "Contents": [
        {
            "Key": "file_with_data",
            "LastModified": "2020-10-04T15:47:59+00:00",
            "ETag": "\"2e237c53edcc537f2e12ad1b0e613b0e\"",
            "Size": 4692,
            "StorageClass": "STANDARD",
            "Owner": {
                "DisplayName": "nbecker",
                "ID": "6ad7c630ff458fbcf3fc06c2625a64359f33904bf84c4e6531a3a59d94557d08"
            }
        }
    ],
    "Name": "jenia",
    "Prefix": "",
    "MaxKeys": 1,
    "EncodingType": "url"
}
`
While in the list of blob we would get a key.
I know that this is not a part of your change, but this seems like a bug.
We can open an issue and handle it later if you'd like.

Reply:
of course, it's not compatible to list 1 result instead of 0 but it's that way for not returning Bad Request error.
I can check for maxResults ==0 before like we do today and return empty values like AWS does instead of checking maxResults to 1. WDYT?

Reply:
In my opinion, it is better not to return values on 0 (even though Azure doesn't allow such requests to be sent), rather than returning values that the user did not request.
The second option is just to throw an exception from Azure NS and our NS merge will handle that.
If we'd have a merge between AWS and Azure then we will get a partial response from AWS, but if we will have a single Azure NS then we will throw an exception.
There are other variants as well, but I think these are the main two.

Reply:
@guymguym 
what was the reason to change the limit to 1 in the first place instead of just returning empty values?
We decided to go with the first option instead if you don't have counterclaims